wiring/inc/spark_wiring_ble.h

```diff
@@ -421,114 +457,164 @@ class BlePeerDevice {
     BlePeerDevice();
     ~BlePeerDevice();
 
-    BleCharacteristic getCharacteristicByDescription(const char* desc);
-    BleCharacteristic getCharacteristicByDescription(const String& desc);
-    BleCharacteristic getCharacteristicByUUID(const BleUuid& uuid);
+    // Discover all services on peer device.
+    Vector<BleService>& discoverAllServices();
```

Comment:
These objects use shared pointer so they are copyable. By returning a reference it can save time copying the objects. If the time doesn't matter, we can absolutely simply return by value.

Reply:
The semantics aren't clear though in this case. If I see something returned by reference, I'd say I'm freely allowed to modify it, which shouldn't be the case for these methods.